src/inner-slider.js

```diff
@@ -88,7 +88,7 @@ export var InnerSlider = React.createClass({
           index: nextProps.slickGoTo,
           currentSlide: this.state.currentSlide
       });
-    } else if (this.state.currentSlide >= nextProps.children.length) {
+    } else if (this.state.currentSlide >= React.Children.count(nextProps.children)) {
```

Comment:
Hi. What is with the "nextProps.children.length" in the changeSlide Method (line 95) ? It should be fixed, too.